File: Problems/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int x, y, i = m - 1, j = n - 1, k  = m + n - 1;
        for(;i >= 0 && j >= 0;) {
            x = nums1[i];
            y = nums2[j];
            if(x > y) {
                nums1[k] = x;
                k--;
                i--;
            } else {
                nums1[k] = y;
                k--;
                j--;
            }
        }
        for(;j >= 0;)  {
            nums1[k] = nums2[j];
            j--;
            k--;
        }
        for(;i >= 0;) {
            nums1[k] = nums1[i];
            i--;
            k--;
        }
    }
}

File: Problems/merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        x = m-1
        y = n-1
        for i in reversed(range(0,m+n)):
            if x<0 or y<0:
                break
            if nums1[x] < nums2[y]:
                nums1[i] = nums2[y]
                y -= 1
            else:
                nums1[i] = nums1[x]
                x -= 1
        
        if x >=0:
            for i in reversed(range(0,x+1)):
                nums1[i] = nums1[i]
        if y >=0:
            for i in reversed(range(0,y+1)):
                nums1[i] = nums2[i]
